Bottom: 45fa470a0712f0a79d131d4d3fc0c792e0147118
Top:    834cffa9709212668b8b51540ddd6924f3a4517d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-02-19 15:02:17 +0000

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 8a49422..d43ab45 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1326,7 +1326,7 @@ static int
 clearPad_callback_solid (const BoxType * b, void *cl)
 {
   PadTypePtr pad = (PadTypePtr) b;
-  if (!XOR (TEST_FLAG (ONSOLDERFLAG, pad), SWAP_IDENT))
+  if (!XOR (TEST_FLAG (ONSOLDERFLAG, pad), SWAP_IDENT) && pad->mask)
     ClearPadSolid (pad, true);
   return 1;
 }
